backend/contributions/tests/test_project_milestone_rubric.py

```
@@ -0,0 +1,612 @@
+            },
+            content_type='application/json',
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertFalse(ProjectMilestoneReview.objects.filter(submitted_contribution=submission).exists())
+
+    def test_project_review_model_enforces_builder_project_only(self):
+        project_submission = self.create_submission()
+        standard_submission = self.create_submission(self.standard_type)
+
+        with self.assertRaises(ValidationError):
```

Ruff (0.15.15)

[warning] Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)